packages/core/src/routes/resource.ts

```diff
@@ -95,4 +103,85 @@ export default function resourceRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.get(
+    '/resources/:resourceId/scopes',
+    koaGuard({ params: object({ resourceId: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { resourceId },
+      } = ctx.guard;
+
+      await findResourceById(resourceId);
+      ctx.body = await findScopesByResourceId(resourceId);
+
+      return next();
+    }
+  );
+
+  router.post(
+    '/resources/:resourceId/scopes',
+    koaGuard({
+      params: object({ resourceId: string().min(1) }),
+      body: Scopes.createGuard.pick({ name: true, description: true }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { resourceId },
+        body,
+      } = ctx.guard;
+
+      await findResourceById(resourceId);
+
+      ctx.body = await insertScope({
+        ...body,
+        id: scoupeId(),
+        resourceId,
+      });
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/resources/:resourceId/scopes/:scopeId',
+    koaGuard({
+      params: object({ resourceId: string().min(1), scopeId: string().min(1) }),
+      body: Scopes.createGuard.pick({ name: true, description: true }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { resourceId, scopeId },
+        body,
+      } = ctx.guard;
+
+      await findResourceById(resourceId);
+      await findScopeById(scopeId);
+
+      ctx.body = await updateScopeById(scopeId, body);
```

Comment:
Would it ok if just let the `updateScopeById` throw if no scopeId is found instead of read first? Seems redundant.